easy-ui-react/src/ProductLayout/ProductLayoutHeader.tsx

```diff
@@ -0,0 +1,159 @@
+import MenuIcon from "@easypost/easy-ui-icons/Menu";
+import { CollectionChildren } from "@react-types/shared";
+import React, { Key, ReactNode } from "react";
+import { Button } from "../Button";
+import { Icon } from "../Icon";
+import { Text } from "../Text";
+import { UnstyledButton } from "../UnstyledButton";
+import { classNames } from "../utilities/css";
+import { HelpMenu } from "./HelpMenu";
+import { ScreenSizeSwitcher } from "./ScreenSizeSwitcher";
+import { useProductLayout } from "./context";
+
+import styles from "./ProductLayoutHeader.module.scss";
+
+export type ProductLayoutHeaderProps = {
+  /**
+   * List of help menu items to render. Should be an array of `<Menu.Item />`s.
+   */
+  helpMenuItems: CollectionChildren<object>;
+
+  /**
+   * Handler that is called when a help menu item is selected.
+   */
+  onHelpMenuAction?: (key: Key) => void;
+
+  /**
+   * Primary call to action for the header.
+   */
+  primaryAction?: ProductLayoutHeaderActionProps;
+
+  /**
+   * Logo to render in the header on small screens.
+   */
+  renderSmallScreenLogo: () => ReactNode;
+
+  /**
+   * Secondary call to action for the header.
+   */
+  secondaryAction?: ProductLayoutHeaderActionProps;
+
+  /**
+   * Page title.
+   */
+  title: ReactNode;
+};
+
+export type ProductLayoutHeaderActionProps = {
+  /**
+   * Content text to render for the header action.
+   */
+  content: string;
+
+  /**
+   * Action to invoke for the header action.
+   */
+  onAction: () => void;
+
+  /**
+   * Whether or not the header action is disabled.
+   */
+  isDisabled?: boolean;
+};
+
+export function ProductLayoutHeader(props: ProductLayoutHeaderProps) {
+  return (
+    <header>
+      <ScreenSizeSwitcher
+        renderOnLargeScreen={() => <LargeScreenHeader {...props} />}
+        renderOnSmallScreen={() => <SmallScreenHeader {...props} />}
+      />
```

Comment:
the header felt different enough between large and small screens to render different components instead of trying to rewire them